utils/secret_reader.py

```diff
@@ -49,6 +54,11 @@ def read_all(secret, settings=None):
     """
 
     if settings and settings.get('vault'):
-        return vault_client.read_all(secret)
+        try:
+            data = vault_client.read_all(secret)
+        except hvac.exceptions.Forbidden:
+            raise VaultForbidden(f'permission denied reading vault secret at '
```

Comment:
this will still be retried multiple times. should we be specific in the `retry` function to not catch this?